src/routes/CreateSafePage/components/SafeCreationProcess.tsx

```diff
@@ -78,102 +87,118 @@ const parseError = (err: Error): Error => {
   return actualMessage ? new Error(actualMessage) : err
 }
 
-function SafeCreationProcess(): ReactElement {
-  const [safeCreationTxHash, setSafeCreationTxHash] = useState<string | undefined>()
-  const [creationTxPromise, setCreationTxPromise] = useState<Promise<TransactionReceipt>>()
+const getSavedSafeCreation = (): CreateSafeFormValues | void => {
+  const savedData = loadFromStorage<CreateSafeFormValues>(SAFE_PENDING_CREATION_STORAGE_KEY)
 
-  const { trackEvent } = useAnalytics()
-  const dispatch = useDispatch()
-  const userAddressAccount = useSelector(userAccountSelector)
-  const chainId = useSelector(currentChainId)
+  if (!savedData) {
+    goToWelcomePage()
```

Comment:
I think it's better to keep this function pure and keep the side effects outside of it